### ISSUE TITLE
optimizer ZeroDivisionError

### ISSUE THREAD

**Issue by goerlibe** [discopop-project/discopop#467]
Running the usual steps on this program, then running the optimizer without arguments, i.e. `discopop_optimizer`, results in ZeroDivisionError. Exhaustive Search works fine.

<details><summary>Source Code</summary>

```
#define N 1000

int main() {
    double a[N];
    double b[N];
    for (int i = 0; i < N; ++i) {
        a[i] = (double) i;
    }
    for (int i = 0; i < N *N; ++i) {
        a[i%N] = (double) i;
        b[i%N] = a[i%N] + 42;
    }
    return 0;
}
```

</details> 

<details><summary>Error</summary>

```
Traceback (most recent call last):
  File "/home/bg/.local/bin/discopop_optimizer", line 11, in <module>
    load_entry_point('discopop', 'console_scripts', 'discopop_optimizer')()
  File "/home/bg/dp-workspace/discopop/discopop_library/discopop_optimizer/__main__.py", line 67, in main
    run(arguments)
  File "/home/bg/dp-workspace/discopop/discopop_library/discopop_optimizer/optimizer.py", line 179, in run
    best_configuration = perform_evolutionary_search(
  File "/home/bg/dp-workspace/discopop/discopop_library/discopop_optimizer/optimization/evolutionary_algorithm.py", line 60, in perform_evolutionary_search
    population = __select(
  File "/home/bg/dp-workspace/discopop/discopop_library/discopop_optimizer/optimization/evolutionary_algorithm.py", line 243, in __select
    probabilityMap[tuple(element)] = 1 - (fitness[element_idx] / score_sum)
ZeroDivisionError: float division by zero
```

**Comment by lukasrothenberger** [comment]
Issue has been identified as unrelated to the optimizer.

@goerlibe , please close the issue once you are pleases with your fix in the VSCode Extension.

**Comment by goerlibe** [comment]
Deleting .discopop after running cmake did fix the issue. However a new problem was introduced, because now the Hotspot Detection and DiscoPoP do not play well together. (They each require a new cmake build, therefore deleting each other's results)

my current build process creates the following directories:
- path/to/build
  - .discopop (shared by DiscoPoP and HotspotDetection)
  - DiscoPoP (the build directory for DiscoPoP)
  - HotspotDetection (the build directory for HotspotDetection)

Is it safe to run the cmake command without the DOT_DISCOPOP, then delete the local .discopop directory and run make and the instrumentation with the DOT_DISCOPOP pointing to the above mentioned directory?
Or does some component remember the DOT_DISCOPOP from the cmake command and everything results in a mess?

**Comment by lukasrothenberger** [comment]
"Is it safe to run..."
I think this should be fine.

**Comment by lukasrothenberger** [comment]
Hi @goerlibe,
can close this issue? It should be fixed by #464.
